### PR TITLE
[full-ci] Fix public link loading on role change

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
The loading and resolving of public links when their respective roles have changed has been fixed.
## Related Issue

- Fixes https://github.com/owncloud/web/issues/8003

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic2</strong> https://drone.owncloud.com/owncloud/web/30019/68/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicManagement-sharingPublicSession_feature-L35.png
  

### webUISharingPublicManagement-sharingPublicSession_feature-L48.png
  


</p></details>
:boom: The oCISSharingPublic2 tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/views/spaces/GenericSpace.vue
Comment (JammingBen) [inline packages/web-app-files/src/views/spaces/GenericSpace.vue:341]: Not the biggest fan of putting such logic in the view directly... But the public link is already resolved here, so I don't see another good place to put it.

Comment (kulmann) [reply packages/web-app-files/src/views/spaces/GenericSpace.vue:341]: What about putting it in the `DriveResolver.vue`? 🤔

Comment (JammingBen) [reply packages/web-app-files/src/views/spaces/GenericSpace.vue:341]: This would work as well I think, but do you think it's a better fit though? It's "just another view". Also, the logic only applies to the `GenericSpace` view, not to `GenericTrash` and `DriveRedirect`.

Comment (JammingBen) [reply packages/web-app-files/src/views/spaces/GenericSpace.vue]: I moved it to the DriveResolver. I agree it's a better fit than the already resolved space.

File: packages/web-pkg/src/composables/driveResolver/useDriveResolver.ts
Comment (kulmann) [inline packages/web-pkg/src/composables/driveResolver/useDriveResolver.ts:19]: I'd prefer
```suggestion
interface DriveResolverResult {
```
as we've already used that naming scheme a lot (`...Options` and `...Result`).

File: packages/web-app-files/src/views/spaces/DriveResolver.vue
Comment (dschmidt) [inline packages/web-app-files/src/views/spaces/DriveResolver.vue:60]: What happens if this request fails for whatever reason?

File: packages/web-app-files/tests/unit/views/spaces/DriveResolver.spec.ts
Comment (dschmidt) [inline packages/web-app-files/tests/unit/views/spaces/DriveResolver.spec.ts:48]: I'm afraid we need this? but it's ugly :(

**Review by dschmidt — APPROVED** [review]

File: packages/web-app-files/src/views/spaces/DriveResolver.vue
Comment (JammingBen) [reply packages/web-app-files/src/views/spaces/DriveResolver.vue:60]: The space from `useDriveResolver()` will be used as default. It won't have `publicLinkPermission` though, so the following if-condition will never resolve to `true`.

File: packages/web-app-files/tests/unit/views/spaces/DriveResolver.spec.ts
Comment (JammingBen) [reply packages/web-app-files/tests/unit/views/spaces/DriveResolver.spec.ts:48]: Yes, I didn't find a way around it unfortunately :/

File: packages/web-app-files/src/views/spaces/DriveResolver.vue
Comment (dschmidt) [reply packages/web-app-files/src/views/spaces/DriveResolver.vue:60]: Yeah, my point is: should we show some kind of error?

Comment (JammingBen) [reply packages/web-app-files/src/views/spaces/DriveResolver.vue:60]: Hmm IMO we don't need a user facing error here, as a user would most likely not know what to do with it anyway. But a console error would be good indeed. I'll wait for the review from Benedikt and add it after.

Comment (kulmann) [reply packages/web-app-files/src/views/spaces/DriveResolver.vue:60]: Hm... I think it would make more sense to do the `getFileInfo` before inserting the `PublicSpaceResource` into the store, i.e. inside the `store.watch` of the `web-runtime/src/index.ts`.

**Review by kulmann — CHANGES_REQUESTED** [review]

**Review by kulmann — APPROVED** [review]

File: packages/web-app-files/src/views/spaces/DriveResolver.vue
Comment (kulmann) [reply packages/web-app-files/src/views/spaces/DriveResolver.vue:61]: for the record: this was a can of worms. 😅

**Pull request merged**
